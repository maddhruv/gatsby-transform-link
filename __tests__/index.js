const Remark = require("remark");
const plugin = require("..");

const remark = Remark().use();

describe("Plugin without options", () => {
  it("should render links correctly", () => {
    const markdownAST = remark.parse(`[Example](https://example.com)`);
    plugin({ markdownAST });
    expect(markdownAST).toMatchSnapshot();
  });
});

describe("Plugin with prefix", () => {
  it("should render links correctly", () => {
    const markdownAST = remark.parse(`[Example](https://example.com)`);
    plugin(
      { markdownAST },
      {
        prefix: "ssh+",
      }
    );
    expect(markdownAST).toMatchSnapshot();
  });
});

describe("Plugin with suffix", () => {
  it("should render links correctly", () => {
    const markdownAST = remark.parse(`[Example](https://example.com)`);
    plugin(
      { markdownAST },
      {
        suffix: "?ref=maddhruv",
      }
    );
    expect(markdownAST).toMatchSnapshot();
  });
});

describe("Plugin with both prefix and suffix", () => {
  it("should render links correctly", () => {
    const markdownAST = remark.parse(`[Example](https://example.com)`);
    plugin(
      { markdownAST },
      {
        prefix: "ssh+",
        suffix: "?ref=maddhruv",
      }
    );
    expect(markdownAST).toMatchSnapshot();
  });
});
