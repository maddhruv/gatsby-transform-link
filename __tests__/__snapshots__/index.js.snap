// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plugin with both prefix and suffix should render links correctly 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "position": Position {
                "end": Object {
                  "column": 9,
                  "line": 1,
                  "offset": 8,
                },
                "indent": Array [],
                "start": Object {
                  "column": 2,
                  "line": 1,
                  "offset": 1,
                },
              },
              "type": "text",
              "value": "Example",
            },
          ],
          "position": Position {
            "end": Object {
              "column": 31,
              "line": 1,
              "offset": 30,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "title": null,
          "type": "link",
          "url": "ssh+https://example.com?ref=maddhruv",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 31,
          "line": 1,
          "offset": 30,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 31,
      "line": 1,
      "offset": 30,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`Plugin with prefix should render links correctly 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "position": Position {
                "end": Object {
                  "column": 9,
                  "line": 1,
                  "offset": 8,
                },
                "indent": Array [],
                "start": Object {
                  "column": 2,
                  "line": 1,
                  "offset": 1,
                },
              },
              "type": "text",
              "value": "Example",
            },
          ],
          "position": Position {
            "end": Object {
              "column": 31,
              "line": 1,
              "offset": 30,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "title": null,
          "type": "link",
          "url": "ssh+https://example.com",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 31,
          "line": 1,
          "offset": 30,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 31,
      "line": 1,
      "offset": 30,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`Plugin with suffix should render links correctly 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "position": Position {
                "end": Object {
                  "column": 9,
                  "line": 1,
                  "offset": 8,
                },
                "indent": Array [],
                "start": Object {
                  "column": 2,
                  "line": 1,
                  "offset": 1,
                },
              },
              "type": "text",
              "value": "Example",
            },
          ],
          "position": Position {
            "end": Object {
              "column": 31,
              "line": 1,
              "offset": 30,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "title": null,
          "type": "link",
          "url": "https://example.com?ref=maddhruv",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 31,
          "line": 1,
          "offset": 30,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 31,
      "line": 1,
      "offset": 30,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`Plugin without options should render links correctly 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "position": Position {
                "end": Object {
                  "column": 9,
                  "line": 1,
                  "offset": 8,
                },
                "indent": Array [],
                "start": Object {
                  "column": 2,
                  "line": 1,
                  "offset": 1,
                },
              },
              "type": "text",
              "value": "Example",
            },
          ],
          "position": Position {
            "end": Object {
              "column": 31,
              "line": 1,
              "offset": 30,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "title": null,
          "type": "link",
          "url": "https://example.com",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 31,
          "line": 1,
          "offset": 30,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 31,
      "line": 1,
      "offset": 30,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
